metricbeat/module/elasticsearch/shard/data.go

```diff
@@ -61,13 +60,38 @@ func eventsMapping(r mb.ReporterV2, content []byte) {
 			for _, shard := range shards {
 				event := mb.Event{}
 
-				fields, _ := schema.Apply(shard)
+				fields, err := schema.Apply(shard)
+				if err != nil {
+					r.Error(err)
+					continue
+				}
+
+				// Handle node field: could be string or null
+				err = elasticsearch.PassThruField("node", shard, fields)
+				if err != nil {
+					continue
+				}
+
+				// Handle relocating_node field: could be string or null
+				err = elasticsearch.PassThruField("relocating_node", shard, fields)
+				if err != nil {
+					continue
+				}
+
 				event.ModuleFields = common.MapStr{}
 				event.ModuleFields.Put("node.name", fields["node"])
```

Comment:
@ruflin Should this field be at the module level or the metricset level in the event document? As you can see it is currently at the module level. However, it is not mentioned in https://github.com/elastic/beats/blob/master/metricbeat/module/elasticsearch/_meta/fields.yml#L1-L26

Reply:
It should be on the module level and it works because it's defined here: https://github.com/elastic/beats/blob/master/metricbeat/module/elasticsearch/node/_meta/fields.yml#L7 Not sure where we should put it as part of the module fields.yml or as part of the node fields.yml